src/DI/BackgroundQueueExtension.php

```diff
@@ -101,17 +102,27 @@ public function afterCompile(ClassType $class)
 $service = (function () {
 	'. $serviceMethod->getBody() .'
 })();
+
+$shutdownCallback = function () use ($service) {
+	$service->onShutdown();
+};
+
 if (php_sapi_name() === "cli") {
-	$this->getByType(\Symfony\Component\EventDispatcher\EventDispatcherInterface::class)
-		->addListener(\Symfony\Component\Console\ConsoleEvents::TERMINATE, function () use ($service) {
-			$service->onShutdown();
-		});
+	
+	if (get_class($this->getByType(\Symfony\Component\EventDispatcher\EventDispatcherInterface::class)) === \Kdyby\Events\SymfonyDispatcher::class) {
```

Comment:
proc tady neni instanceof?

Reply:
Šlo tady o ošetření toho, že daná třída `SymfonyDispatcher` postrádá implementaci addListener. Myslím, že intanceof pak nemá moc smysl, protože pokud by někdo chtěl tuhle třídu extendnout a doimplementovat si metodu addListener, pak to bude tuhle novou třídu pořád ignorovat a brát místo ní dále `EventManager`.